google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -256,34 +260,63 @@ def _bqstorage_fetch(self, bqstorage_client):
 
         Args:
             bqstorage_client(\
-                google.cloud.bigquery_storage_v1beta1.BigQueryStorageClient \
+                google.cloud.bigquery_storage_v1.BigQueryReadClient \
             ):
                 A client tha know how to talk to the BigQuery Storage API.
 
         Returns:
             Iterable[Mapping]:
                 A sequence of rows, represented as dictionaries.
         """
-        # NOTE: Given that BQ storage client instance is passed in, it means
-        # that bigquery_storage_v1beta1 library is available (no ImportError).
+        # Hitting this code path with a BQ Storage client instance implies that
+        # bigquery_storage_v1* can indeed be imported here without errors.
+        from google.cloud import bigquery_storage_v1
         from google.cloud import bigquery_storage_v1beta1
 
         table_reference = self._query_job.destination
 
-        read_session = bqstorage_client.create_read_session(
-            table_reference.to_bqstorage(),
-            "projects/{}".format(table_reference.project),
-            # a single stream only, as DB API is not well-suited for multithreading
-            requested_streams=1,
+        is_v1beta1_client = isinstance(
+            bqstorage_client, bigquery_storage_v1beta1.BigQueryStorageClient
         )
 
+        # We want to preserve comaptibility with the v1beta1 BQ Storage clients,
+        # thus adjust the session creation if needed.
+        if is_v1beta1_client:
+            warnings.warn(
+                "Support for BigQuery Storage v1beta1 clients is deprecated, please "
+                "consider upgrading the client to BigQuery Storage v1 stable version.",
+                category=DeprecationWarning,
+            )
+            read_session = bqstorage_client.create_read_session(
+                table_reference.to_bqstorage(v1beta1=True),
+                "projects/{}".format(table_reference.project),
+                # a single stream only, as DB API is not well-suited for multithreading
+                requested_streams=1,
+            )
+        else:
+            requested_session = bigquery_storage_v1.types.ReadSession(
+                table=table_reference.to_bqstorage(),
+                data_format=bigquery_storage_v1.enums.DataFormat.AVRO,
```

Comment:
I thought our fastpath in python was definitely arrow.  Is there something we're getting from passing through avro?  Or are we in a path where we have the storage import but not arrow?

Reply:
It is ARROW, indeed. This seems like a copy-pasta error, I think I mistakenly took some of the lines from one of the samples that happened to use the AVRO format.

Reply:
Hmm, using ARROW breaks a few system tests, I'll have to check whether this is legitimate or is it just tests that need to be adjusted.

Reply:
Now I remember - I initially chose AVRO, because that's the default the BQ Storage `v1beta1` client uses, and I wanted to match that with the v1 stable client.

When switching to ARROW it turned out that DB API actually never worked with ARROW format, but fortunately system tests caught that. The issue is now fixed and DB API uses ARROW now (both with v1beta1 and v1 stable BQ Storage clients).